### ISSUE TITLE
Abandon Mech instead of ejecting

### ISSUE THREAD

**Issue by BLOODWOLF333** [MegaMek/megamek#161]
If a player's mech say fell into a basement and is all screwed up and you know ejecting the pilot will kill them could we possibly have the option to just abandon the mech like a vehicle?  "exit mech"

**Comment by BLOODWOLF333** [comment]
After talking with Arlith about it, this would make you always want to abandon a mech instead of eject if it is not a case of by skin of the teeth ejection.  So, if a pilot needed to get out immediately he would eject, if the pilot has time (like the mech fell into a basement and is buried) then he could crawl out.  Abandon/exit mech could be declared at the beginning of a turn just like eject but would take a full 10 second turn(take neurohelmet off and various things) and the pilot would not get out till the end phase of that turn or the beginning of the next turn to make it not superior to ejection.  The enemy would have a full turn to pound the mech while the pilot crawls out if it is in open terrain but, would save our pilots from death in bad situations where ejection will surely lead to a pilot death.

**Comment by Akjosch** [comment]
10 seconds is awfully fast to crawl out of battered, broken down, partially buried, possibly laying in a weird angle vehicle's cramped cockpit.

**Comment by BLOODWOLF333** [comment]
True, maybe three turns then (30 sec)?

**Comment by Akjosch** [comment]
Well, three to four turns sounds about right to me. In tabletop, I'd probably leave it at that. We have a few more variables we could easily take into account:

* Cockpit type (normal, torso-mounted, small, industrial, ...), which can have different base values
* Damage to the cockpit location (armour, sensors and internal structure in particular)
* Character values: Hits, Fatigue, Piloting skill, Toughness, Edge
* Character quirks

... and possibly pairing that with a PSR to get out.

Just something to munch on.

**Comment by BLOODWOLF333** [comment]
Yeah I see this could get complex.

- 3 turns to escape base
- +1 turn for cramped and small cockpits.
- +1 turn/every I.S. damage to head location
- add a roll on head crits where a 10+ warps/damages the IS to the point where hatch egress is impossible just like with searchlight destruction on Warhammers
- +1 turn/every pilot hit and point in fatigue
- +1 turn if mech buried under a light building's rubble
- +2 turn if mech buried under a medium building's rubble
- impossible if mech buried under heavy or hardened building rubble (must be dug out)
- if under water(depth 1 only) make a PSR to determine if pilot takes damage when hatch is opened and water comes in
- impossible if mech is in depth 2 water or deeper
- if mech takes damage while upright and pilot is in process of exiting the mech add a (-3) modifier to the roll when determining if pilot takes damage due to the fall.

**Comment by BLOODWOLF333** [comment]
Actually, maybe get rid of the +1 turn/every IS damage and just leave the "egress impossible on a 10+".

**Comment by arlith** [comment]
> 10 seconds is awfully fast to crawl out of battered, broken down, partially buried, possibly laying in a weird angle vehicle's cramped cockpit.

I agree, but one potential issue is TO allowing vehicle crews to abandon vehicles.  Basically, these function along the same rules are ejecting.  So you could argue that TO rules allow vehicle crews to abandon vehicles in 10 seconds, why can't a Mechwarrior get out in 10 seconds?

**Comment by BLOODWOLF333** [comment]
You could bring up that a mechwarrior has the neurohelmet with all the wires and connections and coolant vest with all of its hoses.  Going off the novels.

**Comment by pheonixstorm** [comment]
On this, if you think about it the cockpit will still be blasted off even if the ejection seat isn't working so it wouldn't take a huge amount of time to evac. As for cooling vest and helmet. The hoses should detach easy enough and taking the helmet off shouldn't take long either.

I think the pilot should not be able to crawl out till either the end phase or the begining of the next move phase or just the end phase of the turn after the unit was destroyed

CT cored in turn one, mech abandoned in turn 2.

**Comment by BLOODWOLF333** [comment]
After thinking about this yeah I agree.  Players who have never played the table top game and don't know that a turn is 10 sec long may have issues with it taking 2+ turns for their pilot to exit the mech.  I think they would intuitively pick up on "ok my guy finally passed his PSR to exit the mech and will be outside of it at the beginning of the movement phase next turn".  And probably not have any questions about it.  Exiting a mech can be the exact same roll as a successful ejection check but, failure will require the player to attempt again the next turn until successful.  Which the roll can simulate the pilot fighting with the hatch/canopy and any other hindrances.  Currently MM doesn't apply any modifiers to a mech buried under rubble that fell into a basement and is attempting to stand.  That could be circumnavigated by going hull down which does not require a PSR only two legs, my mech is now upright, exit mech.  I think the "egress impossible on a 10+" could be applied to every head hit regardless of critical hits.  To represent armor, structure, or hinges being damaged/warped or welded by laser fire and making it impossible to exit through the canopy (catapult) or hatch.  All the above still makes ejection a better choice if your mech's armor is gone and your about to get an ammo crit, out in the open, and under fire.  Compared to trying to successfully roll to get out which might fail, and then your stuck there immobile for the rest of the turn, ejection would be better in that example.  But prone or crippled at the bottom of a basement where ejection will kill you, you can just try to crawl out.

**Comment by pheonixstorm** [comment]
Or wait. If your machine is toast and LRMs and Autocannon fire is buzzing over your cockpit wouldn't you feel safer is a dead mech anyway?

**Comment by BLOODWOLF333** [comment]
That option is available currently to keep pilots in their mechs unless that auto abandon rule is turned on.

**Comment by pheonixstorm** [comment]
well it doesn't really work I think. Or at least I haven't had a situation where is didn't work. Every time a mech gets destroyed I always see the eject message.

**Comment by Akjosch** [comment]
You can easily enough simulate "it takes about three turns to exit a shot-up 'Mech in an emergency" by making it a "10+ on 2d6 to exit *ASAP*" roll, with the target number going down by one for each turn you struggle to get out.

Specifically, your chances then are:

* 16.67% to exit on the end of this round (on 10+)
* 23.15% to exit on the end of 2nd round (on 9+, unless you are out already)
* 25.08% to exit on the end of 3rd round (on 8+, unless you are out already)
* 20.48% to exit on the end of 4rd round (on 7+, ...)
* 10.58% to exit on the end of 5th round (on 6+, ...)
* 3.37% to exit on the end of 6th round (on 5+, ...)
* less than 1% for any other result

... for an average of almost exactly 3 rounds (2.98 or so says my quick calculation).

**Comment by BLOODWOLF333** [comment]
Phoenix:  Yeah I honestly have never figured out how to keep a pilot in a downed mech either but, the tooltip for abandon units states it doesnt allow that so, assume you can somehow?

Akjosch: That sounds good to me, keeps it from being a better option than ejection in every case and can also fulfill the time requirement issue.

**Comment by VictorMorson** [comment]
I have a simple suggestion that's largely come from dealing with this (and counting "dead" pilots as not dead if they are abandoning, so this feature would be nice):  Only allow a pilot to abandon their 'mech (rather than make an ejection roll) when the 'mech is shutdown.

Shutdown -> Abandon would likely resolve most of the issues and work out pretty cool.

**Comment by scJazz2** [comment]
I like Akjosch's approach best. but I'd change things up for a "nope it is jammed" option as well. Roll 2 or 3 maybe... yer stuck!

**Comment by kipstafoo** [comment]
Reading through this, bring up something I've thought about for a while -- when a mech is 'dead' why is magically no longer there? It can still take damage. Just because someone ejected/abandoned a unit doesn't mean a unit fusioning in the next hex doesn't rip it apart. IMHO, units should remain on the board until the fight is over. In my TT campaign, many times my players or NPCs have shot dead units while retreating to limit enemy salvage. Maybe this isn't a stock behavior, but having it as an option would be nice.

**Comment by arlith** [comment]
There is a SF bug ticket relating to this issue. https://sourceforge.net/p/megamek/bugs/3885/

**Comment by rjhancock** [comment]
Closing as official rules exist now and tracked in #3716
